zerver/management/commands/delete_realm.py

```diff
@@ -0,0 +1,45 @@
+import os
+import subprocess
+from argparse import ArgumentParser
+from typing import Any
+
+from django.conf import settings
+from django.core.management.base import CommandError
+
+from zerver.lib.management import ZulipBaseCommand
+from zerver.models import Message, UserProfile
+
+
+class Command(ZulipBaseCommand):
+    help = """Script to deactivate a realm."""
+
+    def add_arguments(self, parser: ArgumentParser) -> None:
+        self.add_realm_args(parser, True)
+
+    def handle(self, *args: Any, **options: str) -> None:
+        realm = self.get_realm(options)
+        assert realm is not None  # Should be ensured by parser
+
+        user_count = UserProfile.objects.filter(
+            realm_id=realm.id,
+            is_active=True,
+            is_bot=False,
+        ).exclude(
+            role=UserProfile.ROLE_GUEST,
+        ).count()
+
+        message_count = Message.objects.filter(
+            sender__realm=realm
+        ).count()
+
+        print(f"This realm has {user_count} users and {message_count} messages.\n")
+
+        print("This command will \033[91mPERMANENTLY DELETE\033[0m all data for this realm.  "
+              "Most use cases will be better served by scrub_realm and/or deactivate_realm.")
+
+        confirmation = input("Type the name of the realm to confirm: ")
+        if confirmation != realm.string_id:
+            raise CommandError("Aborting!")
+
+        realm.delete()
+        subprocess.check_call([os.path.join(settings.DEPLOY_ROOT, "scripts", "setup", "flush-memcached")])
```

Comment:
We can merge this as-is and fix forward, but I think we can delete this line if we just add a  `post_delete.connect(flush_realm, sender=Realm)` next to the `post_save`, and tweak the conditionals to make sure we flush everything in the delete case.

@mateuszmandera would you be up for verifying that?

And maybe we should print some sort of output confirming the deletion completed?

Reply:
https://docs.djangoproject.com/en/3.1/ref/signals/#post-delete suggests maybe we can tell it wasn't a save based on `update_fields` not being present as an argument?  One could also imagine just using a lambda to pass our own extra `from_deletion` argument, which might be more readable.